File: 260221/G or H 3/G-or-H-3.py
MAX_NUM = 10000

# 변수 선언 및 입력
n, k = tuple(map(int, input().split()))
arr = [0] * (MAX_NUM + 1)

for _ in range(n):
    x, c = tuple(input().split())
    x = int(x)
    
    arr[x] = 1 if c == 'G' else 2


# 모든 구간의 시작점을 잡아봅니다.
max_sum = 0
for i in range(MAX_NUM - k + 1):
    # 해당 구간 내 원소의 합을 구합니다.
    sum_interval = 0
    for j in range(i, i + k + 1):
        sum_interval += arr[j]

    # 최댓값을 구합니다.
    max_sum = max(max_sum, sum_interval)
                        
print(max_sum)

File: 260221/G or H 3/G-or-H-3.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"math"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())
	scanner.Scan()
	k, _ := strconv.Atoi(scanner.Text())
	pos := make([]int, n)
	letter := make([]string, n)
	maxVal := math.MinInt
	for i := 0; i < n; i++ {
		scanner.Scan()
		posVal, _ := strconv.Atoi(scanner.Text())
		pos[i] = posVal
		scanner.Scan()
		letter[i] = scanner.Text()
		maxVal = max(maxVal, posVal)
	}

	// Please write your code here.
	line := make([]string, maxVal+1)
	for i := range n {
		line[pos[i]] = letter[i]
	}
	maxVal = math.MinInt
	for i := 1; i < len(line)-k; i++ {
		sum := 0
		for j := range k+1 {
			sum += Score(line[i+j])
		}
		maxVal = max(maxVal, sum)
	}
	fmt.Print(maxVal)
}

func Score(s string) int {
	rst := 0
	if s == "G" {
		rst = 1
	} else if s == "H" {
		rst = 2
	}
	return rst
}